src/main/java/ch/njol/skript/expressions/ExprRandomNumber.java

```diff
@@ -35,66 +35,70 @@
 @Examples({"set the player's health to a random number between 5 and 10",
-		"send \"You rolled a %random integer from 1 to 6%!\" to the player"})
-@Since("1.4")
+		"send \"You rolled a %random integer from 1 to 6%!\" to the player",
+		"set {_chances::*} to 5 random integers between 5 and 96"})
+@Since("1.4, INSERT VERSION (Multiple random numbers)")
 public class ExprRandomNumber extends SimpleExpression<Number> {
+
 	static {
 		Skript.registerExpression(ExprRandomNumber.class, Number.class, ExpressionType.COMBINED,
-				"[a] random (1¦integer|2¦number) (from|between) %number% (to|and) %number%");
+				"[a|%-number%] random (:integer|number)[s] (from|between) %number% (to|and) %number%");
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<? extends Number> lower, upper;
-	
+
+	private Expression<Number> amount, lower, upper;
 	private final Random rand = new Random();
-	
 	private boolean integer;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
-		lower = (Expression<Number>) exprs[0];
-		upper = (Expression<Number>) exprs[1];
-		integer = parser.mark == 1;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		amount = (Expression<Number>) exprs[0];
+		lower = (Expression<Number>) exprs[1];
+		upper = (Expression<Number>) exprs[2];
+		integer = parseResult.hasTag("integer");
 		return true;
 	}
-	
+
 	@Override
 	@Nullable
-	protected Number[] get(final Event e) {
-		final Number l = lower.getSingle(e);
-		final Number u = upper.getSingle(e);
-		if (u == null || l == null)
-			return null;
-		final double ll = Math.min(l.doubleValue(), u.doubleValue());
-		final double uu = Math.max(l.doubleValue(), u.doubleValue());
+	protected Number[] get(Event event) {
+		Number lowerNumber = lower.getSingle(event);
+		Number upperNumber = upper.getSingle(event);
+		if (upperNumber == null || lowerNumber == null)
+			return new Number[0];
+		int amount = this.amount == null ? 1 : this.amount.getOptionalSingle(event).orElse(1).intValue();
+		double lower = Math.min(lowerNumber.doubleValue(), upperNumber.doubleValue());
+		double upper = Math.max(lowerNumber.doubleValue(), upperNumber.doubleValue());
 		if (integer) {
-			return new Long[] {Math2.ceil(ll) + Math2.mod(rand.nextLong(), Math2.floor(uu) - Math2.ceil(ll) + 1)};
+			Long[] longs = new Long[amount];
+			for (int i = 0; i < amount; i++)
+				longs[i] = Math2.ceil(lower) + Math2.mod(rand.nextLong(), Math2.floor(upper) - Math2.ceil(lower) + 1);
+			return longs;
 		} else {
-			return new Double[] {ll + rand.nextDouble() * (uu - ll)};
+			Double[] doubles = new Double[amount];
+			for (int i = 0; i < amount; i++)
+				doubles[i] = lower + rand.nextDouble() * (upper - lower);
```

Comment:
This is not guaranteed to be within the bounds due to number rounding, you should check whether the new value is less than upper.

Reply:
i see, this is also another thing that was unchanged by me, presuming i just need to make sure the bigger one is under upper?